### PR TITLE
Upgrade test framework versions and fix test issues

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of https://github.com/aspnet/Coherence-Signed/issues/497.

Update to xunit 2.3 and vstest.console 15.3. 
Fix up any errors found by xunit.analyzers.

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs [hunk 1]
```diff
@@ -159,7 +159,7 @@ public HubConnectionTests()
                     var ex = await Assert.ThrowsAnyAsync<Exception>(
                         async () => await connection.Invoke<object>("!@#$%"));
 
-                    Assert.Equal(ex.Message, "Unknown hub method '!@#$%'");
+                    Assert.Equal("Unknown hub method '!@#$%'", ex.Message);
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs]: SO HAPPY FOR THIS ANALYZER

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs]: 👏  @marcind

Comment (marcind) [inline test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs]: It aims to provide marginal value.

**Comment by natemcmaster** [comment]
@anurse `Microsoft.Extensions.WebSockets.Internal.Tests.WebSocketConnectionTests.AutomaticPingTransmission` failed in AppVeyor. Known issue?

**Comment by anurse** [comment]
That test has been flaky in the past, but I thought we had deflaked it relatively well. Let's see what the rebuild I just started does.

**Pull request merged**